alchemist-mix.el

```diff
@@ -154,22 +167,29 @@
   (interactive "Mmix local.install URL: ")
   (alchemist-mix-execute (list "local.install" url)))
 
-(defun alchemist-mix-hex-search (command)
-  "Display packages matching the given search query."
-  (interactive "Mmix hex.search: ")
-  (alchemist-mix-execute (list "hex.search" command)))
-
-(defun alchemist-mix-help (command)
-  "Show help output for a specific mix command."
-  (interactive "Mmix help: ")
-  (alchemist-mix-execute (list "help" command)))
-
-(defun alchemist-mix-execute (cmdlist)
-  "Run a mix command."
-  (interactive "Mmix: ")
-  (let ((old-directory default-directory))
+(defun alchemist-mix-hex-search (command &optional prefix)
+  "Display packages matching the given search query. Prompt for the mix env
+if the prefix arg is set."
+  (interactive "Mmix hex.search: \nP")
+  (alchemist-mix-execute (list "hex.search" command) prefix))
+
+(defun alchemist-mix-help (command &optional prefix)
+  "Show help output for a specific mix command. Prompt for the mix env if
+the prefix arg is set."
+  (interactive "Mmix help: \nP")
```

Comment:
Why is the `\n` added before the `P`? I stuck it in there because I thought it's used to separate the character codes. Could be wrong tho, still new to this elisp stuff.